contrib/swarm/docker-compose.yml

```diff
@@ -0,0 +1,99 @@
+# This file allow you to run mattermost within your docker swarm mode cluster
+# for more informations check: https://docs.docker.com/engine/swarm/
+#
+# Simply run:
+#
+# `docker stack up [STACK NAME] -c docker-compose.yml`
+# 
+# In this case `mm` is going to be stack name, so the command will be:
+#
+# `docker stack up mm -c docker-compose.yml`
+#
+# From now on all the services that belong to this stack will be prefixed with `mm_`
+# this file defines 3 services, these are going to be mm_db, mm_app and mm_web,
+# each of these names is the service's hostname as well, they can communicate 
+# with each other easily by using the hostname instead of the ip or exposing ports to the host.
+#
+# As a side note, images tagged as latest are pulled by default, 
+# that means there's no need to use `image:latest`
+#
+# use latest compose v3.3 file format for optimal compatibility with latest docker release and swarm features.
+# see https://docs.docker.com/compose/compose-file/compose-versioning/#version-3
+# and https://docs.docker.com/compose/compose-file/compose-versioning/#version-33
+# and https://docs.docker.com/compose/compose-file/compose-versioning/#upgrading
+
+version: '3.3'
+networks:
+  mm-network:
+volumes:
+  mm-dbdata:
+services:
+  db:
+    # use official mattermost prod-db image
+    image: mattermost/mattermost-prod-db
+    networks:
+      - mm-network
+    volumes:
+      # use a named-volume for data persistency
+      - mm-dbdata:/var/lib/postgresql/data
+      - /etc/localtime:/etc/localtime:ro
+    environment:
+      - POSTGRES_USER=mmuser
+      - POSTGRES_PASSWORD=mmuser_password
+      - POSTGRES_DB=mattermost
+      # uncomment the following to enable backup
+      # - AWS_ACCESS_KEY_ID=XXXX
+      # - AWS_SECRET_ACCESS_KEY=XXXX
+      # - WALE_S3_PREFIX=s3://BUCKET_NAME/PATH
+      # - AWS_REGION=us-east-1
+    deploy:
+      restart_policy:
+        condition: on-failure
+  app:
+    # use official mattermost prod-app image
+    image: mattermost/mattermost-prod-app
+    networks:
+      - mm-network
+    volumes:
+      - /var/lib/mattermost/config:/mattermost/config:rw
+      - /var/lib/mattermost/data:/mattermost/data:rw
+      - /var/lib/mattermost/logs:/mattermost/logs:rw
+      - /etc/localtime:/etc/localtime:ro
+    environment:
+      # use service's hostname
+      - DB_HOST=mm_db
+      # talk to the port within the overlay network
+      # without (over)exposing ports
+      - DB_PORT_NUMBER=5432
+      - MM_USERNAME=mmuser
+      - MM_PASSWORD=mmuser_password
+      - MM_DBNAME=mattermost
+      # pass the edition to be used, default is enterprise
+      # setting this env var will make the app use the team edition
+      - edition=team
+      # in case your config is not in default location
+      # - MM_CONFIG=/mattermost/config/config.json
+    deploy:
+      restart_policy:
+        condition: on-failure
+  web:
```

Comment:
I've also been able to make mattermost work without the web container provided here, and using traefik instead. Could add an example if wanted.

Reply:
I'm okay to drop this container on the stack file [as we discussed here](https://github.com/mattermost/mattermost-docker/issues/203). I use (and prefer) Traefik too.

Reply:
As mentioned in that thread, this repo may keep flavors with the stack file, one with this web image, another with traefik.

Reply:
As mentioned in that thread, this repo may keep flavors with the stack file, one with this web image, another with traefik.

Reply:
If we can keep both that's a good things, but I don't know how to do that without duplicating a lot of configuration on both files ?

Reply:
Ya, the config duplication is a problem, but I think maintaining both, the current one and the one with traefik will be enough support.

Reply:
Ok, good for me to get both is you want to provide a Traefik example :)